include/moqt/core/transport.h

```diff
@@ -0,0 +1,226 @@
+/*
+ *  Copyright (C) 2024
+ *  Cisco Systems, Inc.
+ *  All Rights Reserved
+ */
+
+#pragma once
+
+#include <quicr/metrics_exporter.h>
+#include <transport/transport.h>
+
+#include <moqt/common.h>
+#include <moqt/config.h>
+#include <moqt/core/messages.h>
+#include <moqt/metrics.h>
+#include <moqt/publish_track_handler.h>
+#include <moqt/subscribe_track_handler.h>
+#include <transport/span.h>
+
+#include <map>
+#include <string>
+#include <string_view>
+
+namespace moq::transport {
+
+    /**
+     * @brief MOQ Implementation supporting both client and server modes
+     * @details MoQ implementation is the handler for either a client or server. It can run
+     *   in only one mode, client or server.
+     */
+    class Transport : public ITransport::TransportDelegate
```

Comment:
wonder if the name of this class being something like MoQSession ?

Reply:
The point here , this is not a Transport either. If we merge actual QUIC transport in this layer, there will be 2 different transport classes. May be Session is not the best name, but Transport is ambigouos too